src/daemon/ca/certauth.rs

```diff
@@ -1108,33 +1125,48 @@ impl CertAuth {
             match self.find_parent_rc(&parent_handle, &parent_rc_name) {
                 Some(rc) => {
                     // We have a matching RC, make requests (note this may be a no-op).
-                    res.append(&mut self.make_request_events(&mut version, ent, rc, signer.deref())?);
+                    event_details.append(&mut self.make_request_events(ent, rc, signer.deref())?);
                 }
                 None => {
                     // Create a resource class with a pending key
-                    let key_id = signer.create_key()?;
+                    let pending_key = signer.create_key()?;
 
-                    let rcn = ResourceClassName::from(next_class_name);
+                    let resource_class_name = ResourceClassName::from(next_class_name);
                     next_class_name += 1;
 
-                    let ns = rcn.to_string();
-
-                    let rc =
-                        ResourceClass::create(rcn.clone(), ns, parent_handle.clone(), parent_rc_name.clone(), key_id);
-                    let rc_add_version = version;
-                    version += 1;
-                    debug!("Updating Entitlements for CA: {}, adding RC: {}", &self.handle, &rcn);
-
-                    let mut request_events = self.make_request_events(&mut version, ent, &rc, signer.deref())?;
-
-                    let added = EvtDet::resource_class_added(&self.handle, rc_add_version, rcn, rc);
-
-                    res.push(added);
-                    res.append(&mut request_events);
+                    let ns = resource_class_name.to_string();
+
+                    let rc = ResourceClass::create(
+                        resource_class_name.clone(),
+                        ns,
+                        parent_handle.clone(),
+                        parent_rc_name.clone(),
+                        pending_key,
+                    );
+
+                    let added = CaEvtDet::ResourceClassAdded {
+                        resource_class_name,
+                        parent: parent_handle.clone(),
+                        parent_resource_class_name: parent_rc_name.clone(),
+                        pending_key,
+                    };
+                    let mut request_events = self.make_request_events(ent, &rc, signer.deref())?;
+
+                    event_details.push(added);
+                    event_details.append(&mut request_events);
                 }
             }
         }
-        Ok(res)
+
+        let mut events = vec![];
+        let mut version = self.version;
+
+        for detail in event_details {
+            events.push(StoredEvent::new(&self.handle, version, detail));
```

Comment:
What are these `StoredEvent`s? Are they added now to reflect changed behaviour from recording events to actually changing stored data?